Frameworks/UIKit/UITextField.mm

```diff
@@ -1183,348 +891,198 @@ - (void)setContentVerticalAlignment:(UIControlContentVerticalAlignment)alignment
-            }
-
-            // no delegate say no, update firstResponder status to be NO
-            strongSelf->_isFirstResponder = NO;
-
-            // more delegate update
-            if ([strongSelf.delegate respondsToSelector:@selector(textFieldDidEndEditing:)]) {
-                [strongSelf.delegate textFieldDidEndEditing:strongSelf];
-            }
-
-            // more notification
-            dispatch_async(dispatch_get_main_queue(), ^{
-                [strongSelf sendActionsForControlEvents:UIControlEventEditingDidEnd];
-                [[NSNotificationCenter defaultCenter] postNotificationName:UITextFieldTextDidEndEditingNotification object:strongSelf];
-            });
-
-            // Update the collapsed/visible state of the cancel button and adjust text size.
-            control.UpdateLayout();
-            [strongSelf _adjustFontSizeToFitWidthOrApplyCurrentFont];
-        }
-    }));
-}
-
-// Handler when KeyDown
-- (void)_setupControlKeyDownHandler:(const Controls::Control&)control {
-    __weak UITextField* weakSelf = self;
+- (void)_ProcessGotFocusEvent {
+    // when GotFocus, check delegate (if exists) to see if it allows start editing
+    if ([self.delegate respondsToSelector:@selector(textFieldShouldBeginEditing:)] && ![self.delegate textFieldShouldBeginEditing:self]) {
+        // delegate says NO, but we already got the focus at this point, need to kill the focus on this control
+        XamlControls::TextFieldKillFocus(self->_textField);
+        return;
+    }
 
-    // hooking up keydown event to process ENTER key
-    control.KeyDown(objcwinrt::callback([weakSelf] (const WF::IInspectable& sender, const Input::KeyRoutedEventArgs& e) {
-        __strong UITextField* strongSelf = weakSelf;
-        if (strongSelf && e.Key() == winrt::Windows::System::VirtualKey::Enter) {
-            BOOL dismissKeyboard = TRUE;
+    // no one says NO, update first responder to YES
+    self->_isFirstResponder = YES;
 
-            // check with delegate if should resign firstResponder and dismiss the keyboard
-            if ([strongSelf.delegate respondsToSelector:@selector(textFieldShouldReturn:)]) {
-                dismissKeyboard = [strongSelf.delegate textFieldShouldReturn:strongSelf];
-            }
+    // more delegate update
+    if ([self.delegate respondsToSelector:@selector(textFieldDidBeginEditing:)]) {
+        [self.delegate textFieldDidBeginEditing:self];
+    }
 
-            if (dismissKeyboard) {
-                [strongSelf resignFirstResponder];
-            }
+    // more notification
+    dispatch_async(dispatch_get_main_queue(), ^{
+        [self sendActionsForControlEvents:UIControlEventEditingDidBegin];
+        [[NSNotificationCenter defaultCenter] postNotificationName:UITextFieldTextDidBeginEditingNotification object:self];
+    });
 
-            e.Handled(true);
-        } else {
-            e.Handled(false);
-        }
-    }));
+    [self _adjustFontSizeToFitWidthOrApplyCurrentFont];
```

Comment:
good point. the behavior on reference platform is the font change while typing, so this needs to moved to textchanged handler.

---
In reply to: [113561500](https://github.com/Microsoft/WinObjC/pull/2616#discussion_r113561500) [](ancestors = 113561500)